### ISSUE TITLE
Unable to create additional cluster security group rules

### ISSUE THREAD

**Issue by johngmyers** [terraform-aws-modules/terraform-aws-eks#1737]
## Description

Attempting to specify `cluster_security_group_additional_rules` results in a "Inconsistent conditional result types" error from Terraform.

## Versions
 - Terraform: v1.0.0
 - Provider(s): 
+ provider registry.terraform.io/hashicorp/aws v3.64.0
+ provider registry.terraform.io/hashicorp/cloudinit v2.2.0
+ provider registry.terraform.io/hashicorp/kubernetes v2.5.0
+ provider registry.terraform.io/hashicorp/local v2.1.0
+ provider registry.terraform.io/hashicorp/tls v3.1.0
+ provider registry.terraform.io/terraform-aws-modules/http v2.4.1
 - Module: 18.0.0

## Reproduction

Add `cluster_security_group_additional_rules` input to module. Run `terraform plan`.

### Code Snippet to Reproduce

```
  cluster_security_group_additional_rules = {
    admin_access = {
      description = "Admin ingress to Kubernetes API"
      cidr_blocks = var.admin_access
      protocol    = "tcp"
      from_port   = 443
      to_port     = 443
      type        = "ingress"
    }
  }
```

## Expected behavior

Additional rule to be added to the created cluster security group.

## Actual behavior

```
Error: Inconsistent conditional result types
│ 
│   on .terraform/modules/eks/main.tf line 122, in resource "aws_security_group_rule" "cluster":
│  122:   for_each = local.create_cluster_sg ? merge(local.cluster_security_group_rules, var.cluster_security_group_additional_rules) : {}
│     ├────────────────
│     │ local.cluster_security_group_rules is object with 3 attributes
│     │ local.create_cluster_sg is true
│     │ var.cluster_security_group_additional_rules is map of object with 1 element
│ 
│ The true and false result expressions must have consistent types. The given expressions are object and object, respectively.
╵
```

## Additional context

`local.cluster_security_group_rules` should probably be a map instead of an object.

**Comment by johngmyers** [comment]
Actually, it's probably the part after the `:` that needs to be a map.

**Comment by antonbabenko** [comment]
This issue has been resolved in version 18.0.1 :tada:
